src/server/scripts/BrokenIsles/zone_mardum.cpp

```diff
@@ -622,6 +625,234 @@ struct npc_illidari_fighting_invasion_begins : public ScriptedAI
+
+            if (!korvas->IsAIEnabled())
+                return;
+
+            korvas->AI()->Talk(SAY_KORVAS_SLAY_MORE_DEMONS, me);
+            me->InterruptNonMeleeSpells(true);
+            me->GetMotionMaster()->MovePath(PATH_KAYN_SUNFURY_NEAR_TELEPORT, false);
+            me->SetAIAnimKitId(ANIM_DH_RUN);
+            me->DespawnOrUnsummon(10s);
+
+            task.Schedule(2s, [this, korvasGuid](TaskContext /*task*/)
+            {
+                Creature* korvas = ObjectAccessor::GetCreature(*me, korvasGuid);
+                if (!korvas)
+                    return;
+
+                korvas->InterruptNonMeleeSpells(true);
+                korvas->GetMotionMaster()->MovePath(PATH_KORVAS_BLOODTHORN_NEAR_TELEPORT, false);
+                korvas->SetAIAnimKitId(ANIM_DH_RUN);
+                korvas->DespawnOrUnsummon(12s);
+            });
+        });
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+private:
+    TaskScheduler _scheduler;
+};
+
+CreatureAI* KaynSunfuryNearLegionBannerAISelector(Creature* creature)
+{
+    if (creature->IsPrivateObject())
+        return new npc_kayn_sunfury_ashtongue_intro_private(creature);
+    return new npc_kayn_sunfury_ashtongue_intro(creature);
+};
+
+class scene_enter_the_illidari_ashtongue : public SceneScript
+{
+public:
+    scene_enter_the_illidari_ashtongue() : SceneScript("scene_enter_the_illidari_ashtongue") { }
+
+    void OnSceneStart(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) override
+    {
+        Creature* sevisObject = GetClosestCreatureWithOptions(player, 30.0f, { .CreatureId = NPC_SEVIS_BRIGHTFLAME_ASHTONGUE, .IgnorePhases = true });
+        if (!sevisObject)
+            return;
+
+        TempSummon* sevisClone = sevisObject->SummonPersonalClone(sevisObject->GetPosition(), TEMPSUMMON_MANUAL_DESPAWN, 0s, 0, 0, player);
+        if (!sevisClone)
+            return;
+
+        sevisClone->CastSpell(player, SPELL_TRACK_TARGET_IN_CHANNEL, false);
+        sevisClone->DespawnOrUnsummon(15s);
+    }
+
+    void OnSceneTriggerEvent(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/, std::string const& triggerName) override
+    {
+        if (triggerName == "SEEFELSABERCREDIT")
+            player->CastSpell(player, SPELL_ASHTONGUE_FELLSABER_KILL_CREDIT, true);
+
+        else if (triggerName == "UPDATEPHASE")
+            PhasingHandler::OnConditionChange(player);
+    }
+};
+
+// 99916 - Sevis Brightflame (Ashtongue Gateway)
+struct npc_sevis_brightflame_ashtongue_gateway_private : public ScriptedAI
+{
+    npc_sevis_brightflame_ashtongue_gateway_private(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        _scheduler.Schedule(1s, [this](TaskContext /*task*/)
+        {
+            Talk(SAY_SEVIS_SAY_FIND_ALLARI, me);
+        });
+
+        _scheduler.Schedule(3s, [this](TaskContext /*task*/)
```

Comment:
same this here as the above, my only problem with scheduler is each lambda must be inside the previous and it creates a lot of indentation if multiple are required